assets/js/controllers/send.controller.js

```diff
@@ -391,6 +391,30 @@ function SendCtrl ($scope, $log, Wallet, Alerts, currency, $uibModal, $uibModalI
     return $q.resolve();
   };
 
+  $scope.handlePaste = (event) => {
+    let dest;
+
+    event.target.value
+      ? dest = event.target.value
+      : dest = $scope.transaction.destinations[0].address;
+
+    $timeout(() => {
+      event.target.value
+        ? dest = event.target.value
+        : dest = $scope.transaction.destinations[0].address;
+
+      function extractFromUri (URI, RegEx, sliceAmount) {
```

Comment:
You may want to move `extractFromUri` into a service or helper, so that it's easier to write tests for it. I believe there is/was already a function somewhere that parses URL's (in that case from QR codes).